WalletWasabi/Wallets/FileSystemBlockRepository.cs

```diff
@@ -314,4 +326,27 @@ private void CreateFolders()
 
 		IoHelpers.EnsureDirectoryExists(BlocksFolderPath);
 	}
+
+	protected override async Task ActionAsync(CancellationToken cancel)
+	{
+		Block[] blocks;
+		lock (BlockQueueLock)
+		{
+			blocks = BlockQueue.ToArray();
+			BlockQueue.Clear();
+		}
+
+		foreach (var block in blocks)
+		{
+			await SaveAsync(block, cancel).ConfigureAwait(false);
```

Comment:
IMO it would be good to have a SaveAsync(blocks), where multiple files could be written out. Usually, we will write more blocks from this point. In this case acquiring the lock will be done only once.